lua/barbar/ui/render.lua

```diff
@@ -215,6 +218,169 @@ local function open_buffers(new_buffers)
   end
 end
 
+--- An incremental animation for `move_buffer_animated`.
+--- @return nil
+local function move_buffer_animated_tick(ratio, current_state)
+  for _, current_number in ipairs(Layout.buffers) do
+    local current_data = state.get_buffer_data(current_number)
+
+    if current_state.running == true then
+      current_data.position = animate.lerp(
+        ratio,
+        (move_animation_data.previous_positions or {})[current_number],
+        (move_animation_data.next_positions or {})[current_number]
+      )
+    else
+      current_data.position = nil
+    end
+  end
+
+  render.update()
+
+  if current_state.running == false then
+    move_animation_data.next_positions = nil
+    move_animation_data.previous_positions = nil
+  end
+end
+
+--- Move a buffer (with animation, if configured).
+--- @param from_idx integer the buffer's original index.
+--- @param to_idx integer the buffer's new index.
+--- @return nil
+function render.move_buffer(from_idx, to_idx)
```

Comment:
Doesn't really fit here but I want all animation code concentrated in a single place, `api.lua` is a bad place to have animation code.

Reply:
This is a breaking change. All functions in API are said to be available to users to use

Reply:
This was a local function.

Reply:
My mistake; I thought you had moved `move_current_buffer`. Marked that comment as resolved.